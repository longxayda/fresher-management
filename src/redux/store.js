import {configureStore} from "@reduxjs/toolkit";
import attendTableOneTraineeSlice from "./attendanceManagementSlice/attendTableOneTraineeSlice";

// import ticketClassSlice from "./attendanceManagementSlice/ticketClassSlice";
import diligentSummarySlice from "./attendanceManagementSlice/diligentSummarySlice";
import leaveRequestSlice from "./attendanceManagementSlice/leaveRequestSlice";
import checkboxFRReducer from "./fresherManageSlice/checkboxFresherSlice";
import evaluationSlice from "./auditorsSlice/evaluationSlice";
import resourceSlice from "./auditorsSlice/resourceSlice";
import auditorSlice from "./auditorsSlice/auditorSlice";
import manageFRReducer from "./fresherManageSlice/fresherManagementSlice";
import bonusPenaltyReducer from "./fresherManageSlice/penaltySlice";

import classSlice from "./classManagementSlice/classSlice";
import classDetailsSlice from "./classManagementSlice/classDetailsSlice";
import learningPathSlice from "./classManagementSlice/learningPathSlice";
import syllabusSlice from "./classManagementSlice/syllabusSlice";
import moduleSlice from "./classManagementSlice/moduleSlice";
import topicSlice from "./classManagementSlice/topicSlice";

import scoreSlice from "./fresherManageSlice/scoreSlice";
import topicMarkSlice from "./fresherManageSlice/topicMarkSlice";
import scheduleSlice from "./auditorsSlice/scheduleSlice";
import createNewAuditSlice from "./auditorsSlice/createNewAuditSlice";
import adminManageAttendanceReducer from "./attendanceManagementSlice/adminManageAttendanceSlice";
import adminTicketReducer from "./attendanceManagementSlice/adminTicketSlice";

import traineeAttendanceTableSlice from "./attendanceManagementSlice/traineeAttendanceTableSlice";
import authSlice from "./authSlice/authSlice";
import roleSlice from "./roleSlice";
import UserListSlice from "./UserListSlices/UserListSlice";
import attendanceSlice from "./attendanceManagementSlice/attendanceSlice";
import traineeUserProfileSlice from "./userProfileSlice/traineeUserProfileSlice";
import adminUserProfileSlice from "./userProfileSlice/adminUserProfileSlice";

import AddNewReportTableSlice from "components/WeeklyReportTables/addNewReportTableSlice";
import DetailReportTableSlice from "components/WeeklyReportTables/detailReportTableSlice";
import ReportListTableSlice from "components/WeeklyReportTables/reportListTableSlice";
import searchSlice from "components/Search/searchSlice";
import ReportModalSlice from "components/ReportModal/reportModalSlice";

import HistoryReportListTableSlice from "components/MonthlyReportTables/historyReportListTableSlice";
import coursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/coursesReportSlice";
import financeReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/financeReportSlice";
import detailCoursesReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/courses/detailCourseReportSlice";
import detailFinanceReportSlice from "../pages/MonthlyReportPage/MonthlyHistoryReport/finance/detailFinanceReportSlice";

import DetailDeliveryTableSlice from "components/MonthlyReportTables/detailDeliveryTableSlice";
import DeliveryReportListTableSlice from "components/MonthlyReportTables/deliveryReportListTableSlice";
import NewDeliveryTableSlice from "components/MonthlyReportTables/newDeliveryTableSlice";

import OutputsReportTableSlice from "components/OutputsReportTables/outputsReportTableSlice";
import fresherReportSlice from "../pages/OutputsReportPage/fresher/fresherReportSlice";
import svttReportSlice from "../pages/OutputsReportPage/svtt/svttReportSlice";

import detailOutputsReportSlice from "../pages/OutputsReportPage/detailOutputsReportSlice";
import detailFresherReportSlice from "../pages/OutputsReportPage/fresher/detailFresherReportSlice";
import detailSVTTReportSlice from "../pages/OutputsReportPage/svtt/detailSVTTReportSlice";

import thunk from "redux-thunk";
const store = configureStore({
	reducer: {
		Filters: searchSlice,
		evaluation: evaluationSlice,
		resource: resourceSlice,
		manageFR: manageFRReducer,
		checkboxFR: checkboxFRReducer,
		attendTableOneTrainee: attendTableOneTraineeSlice,
		// traineeTabTable: TraineeTabTableSlice,
		class: classSlice,
		learningPath: learningPathSlice,
		syllabus: syllabusSlice,
		classDetails: classDetailsSlice,
		module: moduleSlice,
		topicModule: topicSlice,
		auth: authSlice,
		role: roleSlice,
		// UserList
		UserList: UserListSlice,
		schedule: scheduleSlice,
		newAudit: createNewAuditSlice,
		auditor: auditorSlice,

		// attendTable: attendTableSlice,
		// ticketClass: ticketClassSlice,
		diligentSummary: diligentSummarySlice,
		leaveRequest: leaveRequestSlice,
		adminManageAttendance: adminManageAttendanceReducer,
		adminTicket: adminTicketReducer,
		attendance: attendanceSlice,
		traineeAttendanceTable: traineeAttendanceTableSlice,
		bonusPenalty: bonusPenaltyReducer,
		score: scoreSlice,
		topic: topicMarkSlice,
		traineeUserProfile: traineeUserProfileSlice,
		adminUserProfile: adminUserProfileSlice,

		ReportList: ReportListTableSlice,
		AddNewReport: AddNewReportTableSlice,
		DetailReport: DetailReportTableSlice,
		Filters: searchSlice,
		ReportModal: ReportModalSlice,

		HistoryReportList: HistoryReportListTableSlice,
		CoursesReport: coursesReportSlice,
		FinanceReport: financeReportSlice,
		DetailCoursesReport: detailCoursesReportSlice,
		DetailFinanceReport: detailFinanceReportSlice,
		DetailDeliveryReport: DetailDeliveryTableSlice,
		DeliveryReportList: DeliveryReportListTableSlice,
		NewDeliveryReport: NewDeliveryTableSlice,

		OutputsReport: OutputsReportTableSlice,
		FresherReport: fresherReportSlice,
		SVTTReport: svttReportSlice,
		DetailFresherReport: detailFresherReportSlice,
		DetailSVTTReport: detailSVTTReportSlice,
		DetailOutputsReport: detailOutputsReportSlice,
	},
	middleware: [thunk],
});

export default store;
