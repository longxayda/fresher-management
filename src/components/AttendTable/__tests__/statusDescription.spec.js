import React from 'react';
import Enzyme from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });
import store from 'redux/store';
import { Provider } from "react-redux";
import StatusDescription from 'components/AttendTable/StatusDescription';

describe('Test Attendance Management Status Description', () => {
    let component;
    beforeEach(() => {
        component = mount(<Provider store = { store } > <StatusDescription /> </Provider>);
    });

    test('has the correct class', () => {
        expect(component.find('.attendance-status-description').exists()).toEqual(true);
    });

    test('render Status Description\'s component correctly', () => {
        component = mount(<Provider store = { store }> <StatusDescription /> </Provider>);
        expect(component.html()).toMatchSnapshot();
    });

})
