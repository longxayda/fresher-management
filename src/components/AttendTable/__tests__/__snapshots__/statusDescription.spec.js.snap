// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Attendance Management Status Description render Status Description's component correctly 1`] = `" <div class=\\"attendance-status-description\\"><div class=\\"container-fluid\\"><div style=\\"margin: 10px 0px 0px 0px;\\" class=\\"row\\"><div class=\\"col-md-3\\"><div class=\\"row\\"><div class=\\"col-md-2\\"><span class=\\"bg-primary badge bg-primary\\"><h4 style=\\"margin: 0px;\\">P</h4></span></div><div class=\\"col-md-8\\"><h4 style=\\"margin: 0px;\\">Present</h4></div></div></div><div class=\\"col-md-3\\"><div class=\\"row\\"><div class=\\"col-md-2\\"><span class=\\"bg-success badge bg-primary\\"><h4 style=\\"margin: 0px;\\">A</h4></span></div><div class=\\"col-md-8\\"><h4 style=\\"margin: 0px;\\">Absent</h4></div></div></div><div class=\\"col-md-3\\"><div class=\\"row\\"><div class=\\"col-md-2\\"><span class=\\"bg-warning badge bg-primary\\"><h4 style=\\"margin: 0px;\\">L</h4></span></div><div class=\\"col-md-8\\"><h4 style=\\"margin: 0px;\\">Late coming</h4></div></div></div><div class=\\"col-md-3\\"><div class=\\"row\\"><div class=\\"col-md-2\\"><span class=\\"bg-warning badge bg-primary\\"><h4 style=\\"margin: 0px;\\">E</h4></span></div><div class=\\"col-md-8\\"><h4 style=\\"margin: 0px;\\">Early Leaving</h4></div></div></div></div><div style=\\"margin: 10px 0px 0px 0px;\\" class=\\"row\\"><div class=\\"col-md-6\\"><div class=\\"row\\"><div class=\\"col-md-1\\"><span class=\\"bg-secondary badge bg-primary\\"><h4 style=\\"margin: 0px;\\">Ln</h4></span></div><div class=\\"col-md-10\\"><h4 style=\\"margin: 0px;\\">Late coming with no permission</h4></div></div></div><div class=\\"col-md-6\\"><div class=\\"row\\"><div class=\\"col-md-1\\"><span class=\\"bg-secondary badge bg-primary\\"><h4 style=\\"margin: 0px;\\">En</h4></span></div><div class=\\"col-md-10\\"><h4 style=\\"margin: 0px;\\">Early Leaving with no permission</h4></div></div></div></div><div style=\\"margin: 10px 0px 0px 0px;\\" class=\\"row\\"><div class=\\"col-md-6\\"><div class=\\"row\\"><div class=\\"col-md-1\\"><span class=\\"bg-danger badge bg-primary\\"><h4 style=\\"margin: 0px;\\">An</h4></span></div><div class=\\"col-md-10\\"><h4 style=\\"margin: 0px;\\">Absent with no permission</h4></div></div></div></div></div></div> "`;
